src/background.ts

```diff
@@ -551,7 +571,50 @@ async function start() {
   store.set("engineSettings", engineSettings);
 
   await engineManager.runEngineAll();
-  await createWindow();
+}
+
+/**
+ * エンジンの停止とエンジン終了後処理を行う。
+ * 全処理が完了済みの場合 alreadyCompleted を返す。
+ * そうでない場合は Promise を返す。
+ */
+function terminateEngines(): Promise<void> | "alreadyCompleted" {
+  const killingProcessPromises = engineManager.killEngineAll();
+  const numLivingEngineProcess = Object.entries(killingProcessPromises).length;
+
+  // 前処理が完了している場合
+  if (numLivingEngineProcess === 0 && !vvppManager.hasMarkedEngineDirs()) {
+    return "alreadyCompleted";
+  }
+
+  let numEngineProcessKilled = 0;
+
+  // 非同期的にすべてのエンジンプロセスをキル
+  const waitingKilledPromises: Array<Promise<void>> = Object.entries(
```

Comment:
```suggestion
  const waitingKilledPromises: Promise<void>[] = Object.entries(
```

https://qiita.com/KokiSakano/items/d9ba5f65cf8e64bf8ad7
これを入れても良いかも？

Reply:
typescriptのeslintの`stylistic`に含まれてるっぽいですね！
type checkingをしないなら含めたほうが良いと書いてました。気が向いたら導入してみます！
https://typescript-eslint.io/linting/configs/#projects-without-type-checking

Reply:
これはtypescript-eslintの6.0.0メジャーアップデート後に導入されたもので、かつvue-typescript-eslintは未対応だったので導入できませんでした 😇 

- https://github.com/vuejs/eslint-config-typescript/issues/57